### PR TITLE
[AIRFLOW-3962] Added graceful handling for creation of dag_run of a dag which doesn't have any task

### PR DESCRIPTION
**Pull request by vardancse**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3962
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Scheduler throwing an exception while creating dag_run for a dag which doesn't contain any task, this PR includes graceful Han doing by adding a check.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/jobs.py [hunk 1]
```diff
@@ -865,8 +865,8 @@ def create_dag_run(self, dag, session=None):
                     dag.start_date, next_run_date
                 )
 
-            # don't ever schedule in the future
-            if next_run_date > timezone.utcnow():
+            # don't ever schedule in the future or if next_run_date is None
+            if not next_run_date or next_run_date > timezone.utcnow():
```
Comment (ashb) [inline airflow/jobs.py:869]: When is next run date None? Can we add a unit test for this?

Comment (vardancse) [reply airflow/jobs.py:869]: [task_start_dates](https://github.com/apache/airflow/blob/f3dfc9495be41221bd08ea753555df0975deb961/airflow/jobs.py#L840) list comes out to be empty if there is no task available in dag's python file, and this cause  next_run_date be be None & it has been seen dag.start_date is also coming None.

I'll be adding UT(s) for this change soon.

Comment (vardancse) [reply airflow/jobs.py:869]: @ashb UT has been added, kindly have a look.

File: tests/test_jobs.py [hunk 2]
```diff
@@ -2665,6 +2665,22 @@ def test_scheduler_do_not_schedule_too_early(self):
 
         queue.put.assert_not_called()
 
+    def test_scheduler_do_not_schedule_without_tasks(self):
+        dag = DAG(
+            dag_id='test_scheduler_do_not_schedule_without_tasks',
+            start_date=DEFAULT_DATE)
+
+        session = settings.Session()
```
Comment (Fokko) [inline tests/test_jobs.py]: Please use the `create_session()`

File: tests/test_jobs.py [hunk 3]
```diff
@@ -2665,6 +2665,22 @@ def test_scheduler_do_not_schedule_too_early(self):
...
 
+    def test_scheduler_do_not_schedule_without_tasks(self):
+        dag = DAG(
+            dag_id='test_scheduler_do_not_schedule_without_tasks',
+            start_date=DEFAULT_DATE)
+
+        with create_session() as session:
+            orm_dag = DagModel(dag_id=dag.dag_id)
+            session.merge(orm_dag)
+            session.commit()
```
Comment (Fokko) [inline tests/test_jobs.py]: `.commit()` and `.close()` can be omitted since they are part of the `create_session`

Comment (vardancse) [reply tests/test_jobs.py]: Thanks for catching that, removed suggested lines.

**Comment by vardancse** [comment]
@ashb @Fokko can you have a 2nd review of the PR and let me know the issues, if any?

**Comment by ashb** [comment]
I made a slight tweak to your tests (to reuse the existing session for calls that take it). If the tests pass LGTM.

**Review by ashb — APPROVED** [review]

**Comment by vardancse** [comment]
> I made a slight tweak to your tests (to reuse the existing session for calls that take it). If the tests pass LGTM.

Thanks for looking at it, tests got successful.

File: airflow/jobs.py (hunk 1 above)
Comment (tooptoop4) [reply airflow/jobs.py:869]: @vardancse - next_run_date could be none for purely externally triggered DAGs? Wondering if this change caused https://jira.apache.org/jira/browse/AIRFLOW-4428

Comment (vardancse) [reply airflow/jobs.py:869]: @tooptoop4 AFAIK external triggered request never comes to this route, this is used for creation of scheduled dag_run. Fix was around graceful handling of date time exception, more details available here https://issues.apache.org/jira/browse/AIRFLOW-3962

**Pull request merged**